framework/src/meshgenerators/PatternedMeshGenerator.C

```diff
@@ -88,35 +104,36 @@ PatternedMeshGenerator::generate()
                  type(),
                  " only works with inputs that are replicated.\n\n",
                  "Try running without distributed mesh.");
-    _meshes.push_back(std::move(mesh));
-  }
-
-  // Data structure that holds each row
-  _row_meshes.resize(_pattern.size());
-
-  // Getting the boundaries provided by the user
-  std::vector<BoundaryName> boundary_names = {getParam<BoundaryName>("left_boundary"),
-                                              getParam<BoundaryName>("right_boundary"),
-                                              getParam<BoundaryName>("top_boundary"),
-                                              getParam<BoundaryName>("bottom_boundary")};
+    _meshes[i] = dynamic_pointer_cast<ReplicatedMesh>(mesh->clone());
 
-  std::vector<boundary_id_type> ids =
-      MooseMeshUtils::getBoundaryIDs(*_meshes[0], boundary_names, true);
+    const auto ids = MooseMeshUtils::getBoundaryIDs(*_meshes[i], boundary_names, false);
```

Comment:
In order to properly trigger the paramError for "inputs", I changed the "generate_unknown" parameter in getBoundaryIDs from true to false. When true, a new boundary_id is generated if no match to the requested boundary name is found. When false, invalid_id is returned, which is what triggers the paramError below.

By changing generate_unknown to False, so of the tests in the reactor module are failing because the paramError is being triggered. My hypothesis is that these tests rely on new boundary ids being generated in a way that we are not taking into account.

Reply:
FYI @miaoyinb and @yjung-anl, looks like we've got a pattern stitch that relies on some undefined behavior that we cleaned up (see https://civet.inl.gov/job/1331932/)

Reply:
The failing tests have been updated to work correctly.